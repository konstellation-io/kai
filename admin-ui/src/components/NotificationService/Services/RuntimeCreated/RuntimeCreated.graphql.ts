import gql from 'graphql-tag';

export const RUNTIME_CREATED_SUBSCRIPTION = gql`
  subscription runtimeCreated {
    commentAdded(repoFullName: $repoFullName) {
      id
      name
    }
  }
`;
