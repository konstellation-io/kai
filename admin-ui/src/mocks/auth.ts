import { GET_USER_EMAIL } from '../components/Header/Header.graphql';

export const usernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'user@konstellation.com'
      }
    }
  }
};

export const unauthorizedUsernameMock = {
  request: {
    query: GET_USER_EMAIL
  },
  result: {
    data: {
      me: {
        email: 'unauthorizedUser@konstellation.com'
      }
    }
  },
  error: new Error('Unauthorized user')
};
