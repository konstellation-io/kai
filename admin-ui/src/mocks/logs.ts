import { GET_LOGS } from '../graphql/client/queries/getLogs.graphql';

export const currentLogPanelMock = {
  request: {
    query: GET_LOGS
  },
  result: {
    data: {
      runtimeId: 'runtimeIdMock',
      nodeId: 'nodeIdMock',
      nodeName: 'nodeNameMock'
    }
  }
};
