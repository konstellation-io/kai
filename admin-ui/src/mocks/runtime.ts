import { GET_DASHBOARD } from '../pages/Dashboard/Dashboard.graphql';
import { ADD_RUNTIME } from '../pages/AddRuntime/AddRuntime.graphql';

export const dashboardMock = {
  request: {
    query: GET_DASHBOARD
  },
  result: {
    data: {
      dashboard: {
        runtimes: [
          {
            id: '00001',
            status: 'ACTIVE',
            name: 'Some Name',
            creationDate: '2018-01-02'
          },
          {
            id: '00002',
            status: 'ACTIVE',
            name: 'Some Other Name',
            creationDate: '2018-02-03'
          }
        ],
        alerts: [
          {
            id: 'id0',
            type: 'error',
            message: 'some message',
            runtime: { id: '00001' }
          },
          {
            id: 'id1',
            type: 'error',
            message: 'some message 2',
            runtime: { id: '00001' }
          }
        ]
      }
    }
  }
};

export const addRuntimeMock = {
  request: {
    query: ADD_RUNTIME,
    variables: { name: 'New Runtime' }
  },
  result: {
    data: {
      createRuntime: {
        success: true,
        message: 'some message'
      }
    }
  }
};
