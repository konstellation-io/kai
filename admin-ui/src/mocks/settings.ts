import {
  GET_DOMAINS,
  GET_EXPIRATION_TIME,
  ADD_ALLOWED_DOMAIN,
  UPDATE_EXPIRATION_TIME
} from '../pages/Settings/Settings.graphql';
import { GET_USERS_ACTIVITY } from '../components/UserActivityList/UserActivityList.graphql';

export const domainMock = {
  request: {
    query: GET_DOMAINS
  },
  result: {
    data: {
      settings: {
        authAllowedDomains: ['domain.1', 'domain.2', 'domain.3.sample']
      }
    }
  }
};

export const usersActivityMock = {
  request: {
    query: GET_USERS_ACTIVITY
  },
  result: {
    data: {
      usersActivity: [
        {
          user: 'user1@domain.com',
          message: 'some message 1',
          date: '2019-01-01'
        },
        {
          user: 'user2@domain.com',
          message: 'some message 2',
          date: '2019-01-02'
        },
        {
          user: 'user3@domain.com',
          message: 'some message 3',
          date: '2019-01-03'
        }
      ]
    }
  }
};

export const expirationTimeMock = {
  request: {
    query: GET_EXPIRATION_TIME
  },
  result: {
    data: {
      settings: {
        cookieExpirationTime: 45
      }
    }
  }
};

export const addAllowedDomainMock = {
  request: {
    query: ADD_ALLOWED_DOMAIN,
    variables: { domainName: 'intelygenz.com' }
  },
  result: {
    data: {
      addAllowedDomain: {
        authAllowedDomains: [
          'domain.1',
          'domain.2',
          'domain.3.sample',
          'intelygenz.com'
        ]
      }
    }
  }
};

export const removeAllowedDomainMock = {
  request: {
    query: ADD_ALLOWED_DOMAIN,
    variables: { domainName: 'domain.3.sample' }
  },
  result: {
    data: {
      removeAllowedDomain: {
        authAllowedDomains: ['domain.1', 'domain.2']
      }
    }
  }
};

export const updateExpirationTime = {
  request: {
    query: UPDATE_EXPIRATION_TIME,
    variables: { input: { cookieExpirationTime: 10 } }
  },
  result: {
    data: {
      setSettings: {
        cookieExpirationTime: 12
      }
    }
  }
};
