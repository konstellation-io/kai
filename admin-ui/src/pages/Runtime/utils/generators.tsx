import React from 'react';
import Button from '../../../components/Button/Button';
import StartIcon from '@material-ui/icons/SkipNext';
import StopIcon from '@material-ui/icons/PauseCircleFilled';
import PublishIcon from '@material-ui/icons/PlayCircleFilledOutlined';
import UnpublishIcon from '@material-ui/icons/Block';
import { VersionStatus } from '../../../graphql/types/globalTypes';

function generateActionButton(label: string, icon: any, action: Function) {
  return (
    <Button
      key={label}
      label={label}
      onClick={action}
      height={30}
      Icon={icon}
    />
  );
}

function getStateToButtons(
  publishAction: Function,
  startAction: Function,
  stopAction: Function,
  unpublishAction: Function
) {
  const buttonStart = generateActionButton('START', StartIcon, startAction);
  const buttonStop = generateActionButton('STOP', StopIcon, stopAction);
  const buttonPublish = generateActionButton(
    'PUBLISH',
    PublishIcon,
    publishAction
  );
  const buttonUnpublish = generateActionButton(
    'UNPUBLISH',
    UnpublishIcon,
    unpublishAction
  );

  return {
    [VersionStatus.STOPPED]: [buttonStart],
    [VersionStatus.PUBLISHED]: [buttonUnpublish],
    [VersionStatus.STARTED]: [buttonPublish, buttonStop],
    [VersionStatus.STOPPED]: [buttonStart]
  };
}

export function getVersionActionButtons(
  publishAction: Function,
  startAction: Function,
  stopAction: Function,
  unpublishAction: Function,
  status?: string
) {
  const stateToButtons: { [key: string]: any } = getStateToButtons(
    publishAction,
    startAction,
    stopAction,
    unpublishAction
  );

  return stateToButtons[status || ''];
}
