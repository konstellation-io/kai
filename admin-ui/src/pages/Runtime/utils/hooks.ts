import { useMutation } from '@apollo/react-hooks';
import {
  ACTIVATE_VERSION,
  DEACTIVATE_VERSION,
  DEPLOY_VERSION,
  STOP_VERSION,
  ActivateVersionResponse,
  DeactivateVersionResponse,
  DeployVersionResponse,
  StopVersionResponse,
  VersionActionVars
} from '../pages/RuntimeStatusPreview/RuntimeStatusPreview.graphql';

export enum versionActions {
  activate = 'activateVersion',
  deploy = 'deployVersion',
  stop = 'stopVersion',
  deactivate = 'deactivateVersion'
}

export default function useVersionAction() {
  const [activateMutation, { loading: loadingM1 }] = useMutation<
    ActivateVersionResponse,
    VersionActionVars
  >(ACTIVATE_VERSION);
  const [deactivateMutation, { loading: loadingM2 }] = useMutation<
    DeactivateVersionResponse,
    VersionActionVars
  >(DEACTIVATE_VERSION);
  const [deployMutation, { loading: loadingM3 }] = useMutation<
    DeployVersionResponse,
    VersionActionVars
  >(DEPLOY_VERSION);
  const [stopMutation, { loading: loadingM4 }] = useMutation<
    StopVersionResponse,
    VersionActionVars
  >(STOP_VERSION);

  const mutationLoading = [loadingM1, loadingM2, loadingM3, loadingM4].some(
    el => el
  );

  function getMutationVars(versionId: string, comment?: string) {
    const variables = {
      variables: {
        input: {
          versionId: versionId
        }
      }
    } as any;

    if (comment) {
      variables.variables.input.comment = comment;
    }

    return variables;
  }

  return {
    [versionActions.activate]: activateMutation,
    [versionActions.deploy]: deployMutation,
    [versionActions.stop]: stopMutation,
    [versionActions.deactivate]: deactivateMutation,
    getMutationVars,
    mutationLoading
  };
}
