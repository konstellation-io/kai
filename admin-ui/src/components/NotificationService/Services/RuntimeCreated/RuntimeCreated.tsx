import React, { useState } from 'react';
import { useHistory, useLocation } from 'react-router';
import Notification from '../../Notification';
import * as PAGES from '../../../../constants/routes';

import { RUNTIME_CREATED_SUBSCRIPTION } from './RuntimeCreated.graphql';
import { useSubscription } from '@apollo/react-hooks';
import { Runtime } from '../../../../graphql/models';

const NOTIFICATION_TIMEOUT = 15 * 1000;

type Notification = {
  id: string;
  message: string;
};

function RuntimeCreated() {
  const history = useHistory();
  const location = useLocation();

  const [notifications, setNotifications] = useState<Notification[]>([]);

  useSubscription<Runtime>(RUNTIME_CREATED_SUBSCRIPTION, {
    onSubscriptionData: (data: any) => addNotification(data.runtimeCreated)
  });

  function createNotificationObject(runtime: Runtime): Notification {
    return {
      id: runtime.id,
      message: `The RUNTIME "${runtime.name}" has been successfully created!`
    };
  }

  function addNotification(runtime: Runtime) {
    const newNotification = createNotificationObject(runtime);

    // Close notification after 5 seconds
    setTimeout(() => {
      closeNotification(newNotification.id);
    }, NOTIFICATION_TIMEOUT);

    // Refresh dashboard
    if (location.pathname === PAGES.DASHBOARD) {
      history.push(PAGES.DASHBOARD);
    }

    setNotifications(notifications.concat([newNotification]));
  }

  function closeNotification(id: string) {
    const finalNotifications = notifications.filter(
      (notification: Notification) => notification.id !== id
    );

    setNotifications(finalNotifications);
  }

  const notificationComponents = notifications.map(
    (notification: Notification) => (
      <Notification
        message={notification.message}
        buttonLabel="GO TO RUNTIME"
        buttonAction={() => {
          history.push(PAGES.RUNTIME.replace(':runtimeId', notification.id));
          closeNotification(notification.id);
        }}
        onCloseNotification={() => closeNotification(notification.id)}
      />
    )
  );

  return <div>{notificationComponents}</div>;
}

export default RuntimeCreated;
