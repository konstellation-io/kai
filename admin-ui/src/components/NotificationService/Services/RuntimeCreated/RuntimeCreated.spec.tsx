import React from 'react';
import RuntimeCreated from './RuntimeCreated';
import { runtimeCreatedMock } from '../../../../mocks/runtime';
import { MockedProvider } from '@apollo/react-testing';
import { mountApolloComponent } from '../../../../utils/testUtilsEnzyme';

const mocks = [runtimeCreatedMock];

jest.mock('react-router', () => ({
  useLocation: jest.fn(() => ({
    pathname: ''
  })),
  useHistory: jest.fn(() => ({
    pathname: ''
  }))
}));

function Wrapper() {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
      <RuntimeCreated />
    </MockedProvider>
  );
}

describe('RuntimeCreated', () => {
  it('matches snapshot', async () => {
    const wrapper = await mountApolloComponent(<Wrapper />);

    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', async () => {
    const wrapper = await mountApolloComponent(<Wrapper />);

    expect(wrapper.exists('div')).toBeTruthy();
  });

  // TODO: check how to test onSubscriptionData
});
