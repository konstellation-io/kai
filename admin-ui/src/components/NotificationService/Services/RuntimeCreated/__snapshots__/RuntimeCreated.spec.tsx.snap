// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuntimeCreated matches snapshot 1`] = `
<Wrapper>
  <MockedProvider
    addTypename={false}
    mocks={
      Array [
        Object {
          "request": Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "runtimeCreated",
                  },
                  "operation": "subscription",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "runtimeCreated",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                            },
                            Object {
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 71,
                "source": Object {
                  "body": "subscription runtimeCreated {
  runtimeCreated {
    id
    name
  }
}
",
                  "locationOffset": Object {
                    "column": 1,
                    "line": 1,
                  },
                  "name": "GraphQL request",
                },
                "start": 0,
              },
            },
          },
          "result": Object {
            "data": Object {
              "runtimeCreated": Object {
                "id": "some id",
                "name": "some name",
              },
            },
          },
        },
      ]
    }
  >
    <ApolloProvider
      client={
        ApolloClient {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {
                "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                  "id": "some id",
                  "name": "some name",
                },
                "ROOT_SUBSCRIPTION": Object {
                  "runtimeCreated": Object {
                    "generated": true,
                    "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                    "type": "id",
                    "typename": undefined,
                  },
                },
              },
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {
                "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                  "id": "some id",
                  "name": "some name",
                },
                "ROOT_SUBSCRIPTION": Object {
                  "runtimeCreated": Object {
                    "generated": true,
                    "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                    "type": "id",
                    "typename": undefined,
                  },
                },
              },
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": MockLink {
            "addTypename": false,
            "mockedResponsesByKey": Object {
              "{\\"query\\":\\"subscription runtimeCreated {\\\\n  runtimeCreated {\\\\n    id\\\\n    name\\\\n  }\\\\n}\\\\n\\"}": Array [],
            },
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {
                  "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                    "id": "some id",
                    "name": "some name",
                  },
                  "ROOT_SUBSCRIPTION": Object {
                    "runtimeCreated": Object {
                      "generated": true,
                      "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {
                  "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                    "id": "some id",
                    "name": "some name",
                  },
                  "ROOT_SUBSCRIPTION": Object {
                    "runtimeCreated": Object {
                      "generated": true,
                      "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "queryManager": QueryManager {
            "assumeImmutableResults": false,
            "clientAwareness": Object {
              "name": undefined,
              "version": undefined,
            },
            "dataStore": DataStore {
              "cache": InMemoryCache {
                "addTypename": false,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": false,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {
                    "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                      "id": "some id",
                      "name": "some name",
                    },
                    "ROOT_SUBSCRIPTION": Object {
                      "runtimeCreated": Object {
                        "generated": true,
                        "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                        "type": "id",
                        "typename": undefined,
                      },
                    },
                  },
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {
                    "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                      "id": "some id",
                      "name": "some name",
                    },
                    "ROOT_SUBSCRIPTION": Object {
                      "runtimeCreated": Object {
                        "generated": true,
                        "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                        "type": "id",
                        "typename": undefined,
                      },
                    },
                  },
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "fetchQueryRejectFns": Map {},
            "idCounter": 1,
            "inFlightLinkObservables": Map {},
            "link": MockLink {
              "addTypename": false,
              "mockedResponsesByKey": Object {
                "{\\"query\\":\\"subscription runtimeCreated {\\\\n  runtimeCreated {\\\\n    id\\\\n    name\\\\n  }\\\\n}\\\\n\\"}": Array [],
              },
            },
            "localState": LocalState {
              "cache": InMemoryCache {
                "addTypename": false,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": false,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {
                    "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                      "id": "some id",
                      "name": "some name",
                    },
                    "ROOT_SUBSCRIPTION": Object {
                      "runtimeCreated": Object {
                        "generated": true,
                        "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                        "type": "id",
                        "typename": undefined,
                      },
                    },
                  },
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {
                    "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                      "id": "some id",
                      "name": "some name",
                    },
                    "ROOT_SUBSCRIPTION": Object {
                      "runtimeCreated": Object {
                        "generated": true,
                        "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                        "type": "id",
                        "typename": undefined,
                      },
                    },
                  },
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "client": [Circular],
            },
            "mutationStore": MutationStore {
              "store": Object {},
            },
            "onBroadcast": [Function],
            "pollingInfoByQueryId": Map {},
            "queries": Map {},
            "queryDeduplication": true,
            "queryStore": QueryStore {
              "store": Object {},
            },
            "ssrMode": false,
            "transformCache": WeakMap {},
          },
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {
                  "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                    "id": "some id",
                    "name": "some name",
                  },
                  "ROOT_SUBSCRIPTION": Object {
                    "runtimeCreated": Object {
                      "generated": true,
                      "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {
                  "$ROOT_SUBSCRIPTION.runtimeCreated": Object {
                    "id": "some id",
                    "name": "some name",
                  },
                  "ROOT_SUBSCRIPTION": Object {
                    "runtimeCreated": Object {
                      "generated": true,
                      "id": "$ROOT_SUBSCRIPTION.runtimeCreated",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.6.8",
          "watchQuery": [Function],
        }
      }
    >
      <RuntimeCreated>
        <div>
          <Notification
            buttonAction={[Function]}
            buttonLabel="GO TO RUNTIME"
            key="some id"
            message="The RUNTIME \\"some name\\" has been successfully created!"
            onCloseNotification={[Function]}
          >
            <div
              className="container opened"
            >
              <div
                className="message"
                title="The RUNTIME \\"some name\\" has been successfully created!"
              >
                The RUNTIME "some name" has been successfully created!
              </div>
              <Button
                label="GO TO RUNTIME"
                onClick={[Function]}
              >
                <div
                  className="btn default middle label"
                  onClick={[Function]}
                  style={
                    Object {
                      "height": 40,
                      "lineHeight": "40px",
                    }
                  }
                >
                  <span>
                    GO TO RUNTIME
                  </span>
                </div>
              </Button>
              <Button
                Icon={
                  Object {
                    "$$typeof": Symbol(react.memo),
                    "compare": null,
                    "displayName": "CloseIcon",
                    "muiName": "SvgIcon",
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                  }
                }
                label=""
                onClick={[Function]}
              >
                <div
                  className="btn default middle label"
                  onClick={[Function]}
                  style={
                    Object {
                      "height": 40,
                      "lineHeight": "40px",
                    }
                  }
                >
                  <ForwardRef
                    className="icon-regular"
                  >
                    <WithStyles(ForwardRef(SvgIcon))
                      className="icon-regular"
                    >
                      <ForwardRef(SvgIcon)
                        className="icon-regular"
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction",
                            "colorDisabled": "MuiSvgIcon-colorDisabled",
                            "colorError": "MuiSvgIcon-colorError",
                            "colorPrimary": "MuiSvgIcon-colorPrimary",
                            "colorSecondary": "MuiSvgIcon-colorSecondary",
                            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                            "root": "MuiSvgIcon-root",
                          }
                        }
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root icon-regular"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </ForwardRef(SvgIcon)>
                    </WithStyles(ForwardRef(SvgIcon))>
                  </ForwardRef>
                  <span />
                </div>
              </Button>
            </div>
          </Notification>
        </div>
      </RuntimeCreated>
    </ApolloProvider>
  </MockedProvider>
</Wrapper>
`;
